ras_party/views/share_survey_view.py

```diff
@@ -0,0 +1,84 @@
+from ras_party.controllers.validate import Validator, Exists
+from ras_party.views.account_view import auth
+
+share_survey_view = Blueprint('share_survey_view', __name__)
+logger = structlog.wrap_logger(logging.getLogger(__name__))
+
+
+@share_survey_view.before_request
+@auth.login_required
+def before_share_survey_view():
+    pass
+
+
+@auth.get_password
+def get_pw(username):
+    config_username = current_app.config['SECURITY_USER_NAME']
+    config_password = current_app.config['SECURITY_USER_PASSWORD']
+    if username == config_username:
+        return config_password
+
+
+@share_survey_view.route('/share-survey-users-count', methods=['GET'])
+def share_survey_users():
+    """
+    Get total users count who are already enrolled and pending share survey against business id and survey id
+    """
+    payload = request.get_json()
+    try:
+        business_id = payload['business_id']
+        survey_id = payload['survey_id']
+        # this is just to validate that the business_id exists
+        get_business_by_id(business_id)
+    except KeyError:
+        raise BadRequest('Business Id or survey id is missing')
+    response = share_survey_controller.get_users_enrolled_and_pending_share_against_business_and_survey(business_id,
+                                                                                                        survey_id)
+    return make_response(jsonify(response), 200)
+
+
+@share_survey_view.route('/pending-shares', methods=['POST'])
+def post_pending_shares():
+    """
+     Creates new records for pending shares
+     accepted payload example:
+     {  pending_shares: [{
+            "business_id": "business_id"
+            "survey_id": "survey_id",
+            "email_address": "email_address"
+        },
+        {
+            "business_id": "business_id":
+            "survey_id": "survey_id",
+            "email_address": "email_address"
+        }]
+     }
+    """
+    payload = request.get_json() or {}
+    try:
+        pending_shares = payload['pending_shares']
+        if len(pending_shares) == 0:
+            return BadRequest('Payload Invalid')
+        for share in pending_shares:
+            # Validation, curation and checks
+            expected = ('email_address', 'survey_id', 'business_id')
+            v = Validator(Exists(*expected))
+            if not v.validate(share):
+                logger.debug(v.errors)
+                return BadRequest(v.errors)
+        for pending_share in pending_shares:
+            share_survey_controller.pending_share_create(business_id=pending_share['business_id'],
+                                                         survey_id=pending_share['survey_id'],
+                                                         email_address=pending_share['email_address'])
+            # TODO: Add logic to send email
+        return make_response(jsonify({"created": "success"}), 201)
+    except KeyError:
+        return BadRequest('Payload Invalid')
```

Comment:
Can be more specific here too.  This could be something like `pending_shares key not present in payload json`

Reply:
still an invalid payload.

Reply:
made the changes